### PR TITLE
Make lensfun dependency mandatory

### PR DESCRIPTION
**Pull request by sgotti**
This patch makes lensfun a mandatory dependency.

@TurboGit @parafin while doing this I noticed that there's no check on the lensfun version and I was under the impression that current master/0.3.95 lensfun version should be avoided. In the lens iop code there are #ifdef for the 0.3.95 version but this as we discussed [here](https://github.com/darktable-org/darktable/pull/12843#issuecomment-1310097824) it should be avoided.
I think that we should just forbid use of 0.3.95 or try to fix 0.3.95 support (but we should be sure that the output is the same between the two versions or understand how to proceed)

@TurboGit since this is now a mandatory dependency the output was removed from `darktable --version`. Is it right? Should we keep it? Perhaps print the lensfun library version (it isn't done with other optional libaries)?

@TurboGit I have to check if also the CI needs to be updated or it's ok as is.

### CODE REVIEW

**Comment by TurboGit** [comment]
> @TurboGit since this is now a mandatory dependency the output was removed from `darktable --version`. Is it right? Should we keep it? Perhaps print the lensfun library version (it isn't done with other optional libaries)?

Yes, fine.

> I think that we should just forbid use of 0.3.95 or try to fix 0.3.95 support (but we should be sure that the output is the same between the two versions or understand how to proceed)

Agreed, let's just add a pragma error in section where `#define LF_0395` is set with a clear comment that this is broken (on the lensfun part) for now.

**Comment by sgotti** [comment]
> Agreed, let's just add a pragma error in section where `#define LF_0395` is set with a clear comment that this is broken (on the lensfun part) for now.

Done. Though I'm not sure if the message is good enough.

File: src/iop/lens.cc [hunk 1]
```diff
@@ -61,6 +59,7 @@ extern "C" {
 
 #if LF_VERSION == ((0 << 24) | (3 << 16) | (95 << 8) | 0)
 #define LF_0395
+#error lensfun 0.3.95 is not supported since its API is not backward compatible with lensfun stable release.
```
Comment (TurboGit) [inline src/iop/lens.cc:62]: Looks good to me.

**Review by TurboGit — APPROVED** [review]
Works for me, thanks!

**Comment by TurboGit** [comment]
@sgotti : Since you are about to become our lens correction expert may I ask for more?

What about checking if the lensfun has available data for the picture and include the lensfun entry into the combo only when it is the case. I always found frustrating to select the lensfun entry and get a red message saying that I'm out of luck :)

Of course if there is no metadata & no lensfun entry we may just blank out the lens module and display a message as done for mode that do not apply. Like for example the raw chromatic aberrations for a jpeg:

image

Thoughts?

**Comment by TurboGit** [comment]
@sgotti : Just found an issue, strange that I didn't had issue with the regression tests. Anyway, open an image in darkroom, clear history and you'll see that lens correction is enabled by default. Likewise if history clear from lighttable.

**Comment by TurboGit** [comment]
Let me know if you want me to create two issues for the both points above.

**Comment by sgotti** [comment]
> @sgotti : Just found an issue, strange that I didn't had issue with the regression tests. Anyway, open an image in darkroom, clear history and you'll see that lens correction is enabled by default. Likewise if history clear from lighttable.

I'll take a look when I'll be back at home, but I haven't noticed it and I don't remember a change that enables the module by default. Is it possible that you have an auto apply preset?

**Comment by TurboGit** [comment]
> Is it possible that you have an auto apply preset?

Hum... embarrassing :) Yes, I have one created some time ago to test your PR :) You can forget about this one. Sorry for the noise.

**Comment by sgotti** [comment]
> What about checking if the lensfun has available data for the picture and include the lensfun entry into the combo only when it is the case. I always found frustrating to select the lensfun entry and get a red message saying that I'm out of luck :)

I can suppose it was done that way so you are able to manually select another camera/lens.

**Comment by TurboGit** [comment]
> I can suppose it was done that way so you are able to manually select another camera/lens.

Right again... That's not my day :)

**Pull request merged**